File: Homeworks/C#2/Arrays/SelectionSort/SelectionSort.cs
//Sorting an array means to arrange its elements in increasing order. 
//Write a program to sort an array. Use the "selection sort" algorithm: 
//Find the smallest element, move it at the first position, 
//find the smallest from the rest, move it at the second position, etc.

namespace SelectionSort
{
    using System;
    using System.Linq;
    class SelectionSort
    {
        static void Main(string[] args)
        {
            Console.WriteLine("Enter integers separated by a comma: ");
            int[] numbers = Console.ReadLine()
                .Split(new char[] { ' ', ',', '\t' }, StringSplitOptions.RemoveEmptyEntries)
                .Select(x => int.Parse(x))
                .ToArray();

            numbers = Sort(numbers);
            Console.WriteLine(string.Join(", ", numbers));
        }

        private static int[] Sort(int[] numbers)
        {
            int[] sorted = new int[numbers.Length];

            for (int index = 0; index < sorted.Length; index++)
            {
                int tempIndex = Array.IndexOf(numbers, numbers.Min());
                sorted[index] = numbers[tempIndex];
                numbers[tempIndex] = Int32.MaxValue;
            }

            return sorted;
        }
    }
}

File: Homeworks/JS-basics/Arrays/selectionSort.js
// Sorting an array means to arrange its elements in
// increasing order. Write a script to sort an array. Use
// the "selection sort" algorithm: Find the smallest
// element, move it at the first position, find the
// smallest from the rest, move it at the second
// position, etc.

Array.prototype.selectionSort = function(){
    var sorted = [];

    while(this.length) {
        var minInd = 0;
        for (var ind = 0; ind < this.length; ind++) {
            if(this[ind] < this[minInd]) minInd = ind;
        }
        sorted.push(this[minInd]);
        this.splice(minInd, 1);
    }
    this.push(sorted);
}

var numbers =  [8, 12, 3, 4, 5, 2, 11, 13, 7, 4, 15, 14, 12, 8, 1];
numbers.selectionSort();

console.log(numbers.join(', '));
